flowmap/formats/ugrid.py

```diff
@@ -277,55 +275,74 @@ def subgrid(self, t, method):
+        waterlevel_name = self.generate_name(
             self.path,
             suffix='.json',
-            topic=method,
+            topic='waterlevel',
             counter=t
         )
-        logger.info('writing subgrid features')
+        logger.info('writing waterlevels features to {}'.format(waterlevel_name))
         # save featuress
         crs = geojson.crs.Named(
             properties={
                 "name": "urn:ogc:def:crs:EPSG::{:d}".format(self.src_epsg)
             }
         )
         feature_collection['crs'] = crs
-        with open(new_name, 'w') as f:
+        with open(waterlevel_name, 'w') as f:
             geojson.dump(feature_collection, f, cls=CustomEncoder, allow_nan=False, ignore_nan=True)
 
-        if method == 'waterlevel':
-            logger.warn('interpolation not yet implemente')
-        #     logger.info('writing subgrid band')
-        #     # use extreme value as nodata
-        #     try:
-        #         nodata = np.finfo(band.dtype).min
-        #     except ValueError:
-        #         # for ints use a negative value
-        #         nodata = -99999
-        #     options = dict(
-        #         dtype=str(band.dtype),
-        #         nodata=nodata,
-        #         count=1,
-        #         compress='lzw',
-        #         tiled=True,
-        #         blockxsize=256,
-        #         blockysize=256,
-        #         driver='GTiff',
-        #         affine=dem['affine'],
-        #         width=dem['width'],
-        #         height=dem['height'],
-        #         crs=rasterio.crs.CRS({'init': 'epsg:%d' % (self.src_epsg)})
-        #     )
-        #     with rasterio.open(str(new_name), 'w', **options) as dst:
-        #         dst.write(band.filled(nodata), 1)
+        # interpolate waterlevels to grid (this is file based)
+        interpolated_waterlevel_name = self.generate_name(
+            self.path,
+            suffix='.tiff',
+            topic='waterlevel_idw',
+            counter=t
+        )
+        logger.info(
+            'writing interpolated waterlevels to {}'.format(
+                interpolated_waterlevel_name
+            )
+        )
+        subgrid.interpolate_waterlevels(
+            waterlevel_name,
+            interpolated_waterlevel_name,
+            dem,
+            epsg=self.src_epsg
+        )
+        # interpolated_waterlevel_name is now created
+        with rasterio.open(str(interpolated_waterlevel_name)) as ds:
+            interpolated_waterlevel = ds.read(1, masked=True)
+
+        waterdepth_name = self.generate_name(
+            self.path,
+            suffix='.tiff',
+            topic='waterdepth',
+            counter=t
+        )
+        logger.info(
+            'writing waterdepth to {}'.format(
+                waterdepth_name
+            )
+        )
+        # now for the final computation
+        waterdepth = interpolated_waterlevel - dem['band']
+        # save results
```

Comment:
Is it possible to add the folowing line since waterdepths cannot be lower than zero?

```
waterdepth[waterdepth < 0.0] = 0
```